jwst/firstframe/firstframe_step.py

```diff
@@ -0,0 +1,29 @@
+from ..stpipe import Step
+from .. import datamodels
+from . import firstframe_sub
+
+
+class FirstFrameStep(Step):
+    """
+    FirstFrameStep: This is a MIRI specific task.  If the number of groups
+    is greater than 1, the GROUP data quality flags for the first group will
+    be set to DO_NOT_USE.
+    """
+
+    def process(self, input):
+
+        # Open the input data model
+        with datamodels.open(input) as input_model:
+
+            # check the data is MIRI data
+            detector = input_model.meta.instrument.detector
+            if detector[:3] == 'MIR':
+                # Do the firstframe correction subtraction
+                result = firstframe_sub.do_correction(input_model)
+            else:
+                self.log.warning('First Frame Correction is only for MIRI data')
+                self.log.warning('First frame step will be skipped')
+                result = input_model.copy()
+                result.meta.cal_step.firstframe = 'SKIPPED'
```

Comment:
We'll need to add a new entry to datamodels/schemas/core.schema.yaml for this new attribute "firstframe" in the cal_step hierarchy, so that this attribute is defined.

Reply:
Done.  Please check that I did it correctly.